features/netsocket/emac-drivers/TARGET_NXP_EMAC/TARGET_LPC546XX/lpc546xx_emac.cpp

```diff
@@ -0,0 +1,601 @@
+    uint32_t phyAddr = 0;
+    // Get current status
+    PHY_STATE crt_state;
+    bool connection_status;
+
+    PHY_GetLinkStatus(ENET, phyAddr, &connection_status);
+    crt_state.connected = connection_status;
+    // Get the actual PHY link speed
+    PHY_GetLinkSpeedDuplex(ENET, phyAddr, &crt_state.speed, &crt_state.duplex);
+
+    // Compare with previous state
+    if (crt_state.connected != prev_state.connected && emac_link_state_cb) {
+        emac_link_state_cb(crt_state.connected);
+    }
+
+    if (crt_state.speed != prev_state.speed) {
+        uint32_t fes = ENET->MAC_CONFIG;
+        fes &= ~ENET_MAC_CONFIG_FES_MASK;
+        if (prev_state.speed != (phy_speed_t)STATE_UNKNOWN) {
+            fes |= ENET_MAC_CONFIG_FES(!crt_state.speed);
+        } else {
+            fes |= ENET_MAC_CONFIG_FES(crt_state.speed);
+        }
+
+        ENET->MAC_CONFIG = fes;
+    }
+
+    prev_state = crt_state;
+}
+
+bool LPC546XX_EMAC::power_up()
+{
+    /* Initialize the hardware */
+    if (!low_level_init_successful()) {
+        return false;
+    }
+
+    /* Worker thread */
+    thread = create_new_thread("LPC546XX_EMAC_thread", &LPC546XX_EMAC::thread_function, this, THREAD_STACKSIZE, THREAD_PRIORITY, &thread_cb);
+
+    /* Trigger thread to deal with any RX packets that arrived before thread was started */
+    rx_isr();
+
+    /* PHY monitoring task */
+    prev_state.connected = STATE_UNKNOWN;
+    prev_state.speed = (phy_speed_t)STATE_UNKNOWN;
+    prev_state.duplex = (phy_duplex_t)STATE_UNKNOWN;
+
+    phy_task_handle = mbed::mbed_event_queue()->call_every(PHY_TASK_PERIOD_MS, mbed::callback(this, &LPC546XX_EMAC::phy_task));
+
+    /* Allow the PHY task to detect the initial link state and set up the proper flags */
+    osDelay(10);
+
+    /* Active TX/RX. */
+    ENET_StartRxTx(ENET, 1, 1);
+
+    return true;
+}
+
+uint32_t LPC546XX_EMAC::get_mtu_size() const
+{
+    return LPC546XX_ETH_MTU_SIZE;
+}
+
+uint32_t LPC546XX_EMAC::get_align_preference() const
+{
+    return ENET_BUFF_ALIGNMENT;
+}
+
+void LPC546XX_EMAC::get_ifname(char *name, uint8_t size) const
+{
+    memcpy(name, LPC546XX_ETH_IF_NAME, (size < sizeof(LPC546XX_ETH_IF_NAME)) ? size : sizeof(LPC546XX_ETH_IF_NAME));
+}
+
+uint8_t LPC546XX_EMAC::get_hwaddr_size() const
+{
+    return LPC546XX_HWADDR_SIZE;
+}
+
+bool LPC546XX_EMAC::get_hwaddr(uint8_t *addr) const
+{
+    return false;
+}
+
+void LPC546XX_EMAC::set_hwaddr(const uint8_t *addr)
+{
+    memcpy(hwaddr, addr, sizeof hwaddr);
+    ENET_SetMacAddr(ENET, const_cast<uint8_t*>(addr));
+}
+
+void LPC546XX_EMAC::set_link_input_cb(emac_link_input_cb_t input_cb)
+{
+    emac_link_input_cb = input_cb;
+}
+
+void LPC546XX_EMAC::set_link_state_cb(emac_link_state_change_cb_t state_cb)
+{
+    emac_link_state_cb = state_cb;
+}
+
+void LPC546XX_EMAC::add_multicast_group(const uint8_t *addr)
+{
+    /* No-op at this stage */
```

Comment:
When testing with greentea, driver did not pass multicast packets to network stack. You should add set_all_multicast(true) call here, so in case network stack configures multicast (i.e. is ipv6 capable) multicast filtering is disabled.